agent/api/task/task.go

```diff
@@ -445,6 +453,34 @@ func (task *Task) PostUnmarshalTask(cfg *config.Config,
 	return nil
 }
 
+// initNetworkMode initializes/infers the network mode for the task and assigns the result to this task's NetworkMode field.
+// This is necessary because ACS doesn't send the actual network mode declared in task def.
+// TODO [SC]: Instead of doing this, it's perhaps better/easier to just ask ACS to send network mode from task definition. Using this in the meantime to avoid being blocked.
+func (task *Task) initNetworkMode() {
+	if len(task.ENIs) > 0 {
+		task.NetworkMode = AWSVPCNetworkMode
+		return
+	}
+	for _, c := range task.Containers {
+		// ACS sets task def network mode to each and every customer container docker host config, so the presence
+		// of one container with a given network mode is enough to infer the task network mode
+		switch c.GetNetworkModeFromHostConfig() {
+		case BridgeNetworkMode:
+			task.NetworkMode = BridgeNetworkMode
+			return
+		case HostNetworkMode:
+			task.NetworkMode = HostNetworkMode
+			return
+		case "":
+			// the absence of a network mode means the default docker network mode is used, which, for ECS it's bridge.
+			// for future proofing, let's keep inspecting the rest of the containers for this particular case.
```

Comment:
I'm not quite seeing why we want the loop. While it's possible that in the future the default mode could change or what an empty string implies could be different, I think we will still have the guarantee that all containers in the same task payload should have the same network mode (specified or not). If two containers return different values for network mode, or one is specified and the other is not, I would think something has gone wrong. 

I think we can define a `defaultNetworkMode` and return that in case of absence. 

On a separate note, might be a good idea to check if the network mode is valid. I know it's highly unlikely that control plane will send something invalid, but doesn't hurt to verify. Could be useful say when we are testing Agent with mock payload and it has a typo.

Reply:
> I'm not quite seeing why we want the loop.

for example, internal containers have a blank `NetworkConfig`. I could have used `c.IsInternal()` -> `continue` but I think I feel this is safer, were we only declare `bridge` if all containers we know of have blank or empty `NetworkConfig`.

I'm talking to CP so that they send the actual value from task def to avoid this altogether. I think it will happen, so I guess this is ok for now?

Reply:
We are calling `initNetworkMode` first thing in `PostUnmarshalTask`, which is before we add any internal containers and it should be an unmodified list from Control Plane hence I thought they should have consistent network model. But yeah if we are going to get it directly I think that'll be ideal.

Reply:
> We are calling initNetworkMode first thing in PostUnmarshalTask
yeah, although I'm not sure how to test/enforce that in a UT. If someone injects something before this in the future, or even before PostUnmarshalTask, things might change.

Reply:
Makes sense